ducktape/utils/util.py

```diff
@@ -19,26 +19,42 @@
 import time
 
 
-def wait_until(condition, timeout_sec, backoff_sec=.1, err_msg=""):
+def wait_until(condition, timeout_sec, backoff_sec=.1, err_msg="", retry_on_exc=False):
     """Block until condition evaluates as true or timeout expires, whichever comes first.
 
     :param condition: callable that returns True if the condition is met, False otherwise
     :param timeout_sec: number of seconds to check the condition for before failing
     :param backoff_sec: number of seconds to back off between each failure to meet the condition before checking again
     :param err_msg: a string or callable returning a string that will be included as the exception message if the
                     condition is not met
+    :oaram retry_on_exc: if True, will retry if condition raises an exception. If condition raised exception on last
+                         iteration, that exception will be raised as a cause of TimeoutError.
+                         If False and condition raises an exception, that exception will be forwarded to the caller
+                         immediately.
+                         Defaults to False (original ducktape behavior).
```

Comment:
Not sure where you want to put it, but might want to add a TODO with a version number to start tracking breaking changes we might want to make eventually if we ever bump past 1.0

Reply:
Good idea, added a TODO right here; we can simply use `# TODO: [1.0.0]`